ui/src/commonMain/kotlin/com/materiiapps/gloom/ui/widgets/accounts/AccountSwitcherSheet.kt

```diff
@@ -0,0 +1,88 @@
+package com.materiiapps.gloom.ui.widgets.accounts
+
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import cafe.adriel.voyager.navigator.LocalNavigator
+import cafe.adriel.voyager.navigator.currentOrThrow
+import com.materiiapps.gloom.Res
+import com.materiiapps.gloom.ui.components.bottomsheet.BottomSheet
+import com.materiiapps.gloom.ui.components.bottomsheet.BottomSheetLayout
+import com.materiiapps.gloom.ui.components.settings.SettingsButton
+import com.materiiapps.gloom.ui.screens.auth.LandingScreen
+import com.materiiapps.gloom.ui.screens.root.RootScreen
+import com.materiiapps.gloom.ui.utils.navigate
+import com.materiiapps.gloom.ui.viewmodels.settings.AccountSettingsViewModel
+import dev.icerock.moko.resources.compose.stringResource
+import org.koin.androidx.compose.get
+
+@Composable
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalFoundationApi::class)
+fun AccountSwitcherSheet(
+    onDismiss: () -> Unit,
+) {
+    val nav = LocalNavigator.currentOrThrow
+    val viewModel: AccountSettingsViewModel = get()
+    val accounts by remember(viewModel.authManager.accounts) {
+        derivedStateOf {
+            viewModel.authManager.accounts.values
+                .toList()
+                .sortedByDescending { viewModel.authManager.currentAccount?.id == it.id }
```

Comment:
this is still unstable
use immutable lists / immutable wrappers instead (like upvote)